xcodeproj/__init__.py

```diff
@@ -170,6 +170,8 @@ def __setstate__(self, state):
     def _set_weak_refs(self) -> None:
         """Setup the weak references."""
         for obj in self.objects.values():
+            if type(obj) is dict:
```

Comment:
This is just because we aren't deserializing `XCSwiftPackageProductDependency` right? We should deserialize and maintain the current behavior if possible.

Reply:
The goal of this PR is to just unblock our ability to use SPM in Xcode, I'm sorry I should've made that clearer in the title and description. I added this check to avoid having to deserialize `XCSwiftPackageProductDependency` for now but also because I'd prefer if this library didn't break the build on new any new properties in Xcode projects. Feel free to open a new PR to add the new objects if you think users will want to parse those.

Reply:
I guess the real question here is the choice between silently ignoring failures on breaking changes vs explicit exceptions. I always went for the latter since it makes it easier to tell when things have changed and we are less likely to be bitten by it down the road. But, this isn't my project any more so it's your call 😄 

If you do want to merge it, the PR is here: https://github.com/microsoft/xcodeproj/pull/8

Reply:
Thanks Dale!